### PR TITLE
[SPARK-17090][FOLLOW-UP][ML]Add expert param support to SharedParamsCodeGen

### PR DESCRIPTION
**Pull request by hqzizania**
## What changes were proposed in this pull request?

Add expert param support to SharedParamsCodeGen where aggregationDepth a expert param is added.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala [hunk 1]
```diff
@@ -153,6 +154,11 @@ private[shared] object SharedParamsCodeGen {
     } else {
       ""
     }
+    val groupParam = if (param.isExpertParam) {
```
Comment (sethah) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: I think 

````scala
val groupParam = if (param.isExpertParam) "expertParam" else "param"
````
is cleaner. But, super minor detail.

**Comment by sethah** [comment]
LGTM

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala (hunk 1 above)
Comment (hqzizania) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: Thanks, :) , but its next line [methodStr ](https://github.com/apache/spark/pull/14738/files#diff-400ce1c002287d1afbe8b8486a74ecddL156) also needs to be cleaned?

**Comment by srowen** [comment]
Is this the same as https://issues.apache.org/jira/browse/SPARK-17175 ? can you explain this better?

**Comment by hqzizania** [comment]
@srowen this is about a @group name called expertParam,  just a part of [SPARK-17090](https://github.com/apache/spark/pull/14717). SPARK-17175 is for a expert formula which was discussed in https://issues.apache.org/jira/browse/SPARK-17090

**Comment by srowen** [comment]
What JIRA does it relate to? I'm not clear how it relates to either of them.

**Comment by sethah** [comment]
@srowen https://github.com/apache/spark/pull/14717 Implemented a change to add a parameter to linear/logistic regression which allows users to control the aggregation depth on `treeAggregate` when collecting gradient updates. In that PR, we made the `aggregationDepth` param an expert param, but we forgot to update the `sharedParamsCodegen` file to label it as such in the getter. This JIRA is a small patch to modify the codegen file to optionally spit out expert param annotations in the scaladoc.

**Comment by hqzizania** [comment]
Thanks for sethan's comments :)

**Comment by srowen** [comment]
OK, let's link this to the original JIRA then.

**Comment by hqzizania** [comment]
It is linked to SPARK-17090 now. @srowen

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala [hunk 2]
```diff
@@ -167,11 +173,11 @@ private[shared] object SharedParamsCodeGen {
       |
       |  /**
       |   * Param for $doc.
-      |   * @group param
+      |   * @group $groupParam
       |   */
       |  final val $name: $Param = new $Param(this, "$name", "$doc"$isValid)
       |$setDefault
-      |  /** @group getParam */
+      |  /** @group get${groupParam.capitalize} */
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: It should output ```expertGetParam``` if it's ```expertParam```.

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala [hunk 3]
```diff
@@ -397,13 +397,13 @@ private[ml] trait HasAggregationDepth extends Params {
...
    * Param for suggested depth for treeAggregate (>= 2).
-   * @group param
+   * @group expertParam
    */
   final val aggregationDepth: IntParam = new IntParam(this, "aggregationDepth", "suggested depth for treeAggregate (>= 2)", ParamValidators.gtEq(2))
 
   setDefault(aggregationDepth, 2)
 
-  /** @group getParam */
+  /** @group getExpertParam */
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala]: Ditto.

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala (hunk 2 above)
Comment (hqzizania) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: oops...

**Comment by yanboliang** [comment]
It's better to update the PR title as ```[SPARK-17090][FOLLOW-UP][ML] ... ```.

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala (hunk 1 above)
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: It's better to name it as ```groupStr``` or ```groupAnnotation```.

**Comment by yanboliang** [comment]
LGTM except the last minor issue. Thanks!

**Comment by hqzizania** [comment]
Fixed. Thanks for the reviews.

**Comment by yanboliang** [comment]
Merged into master. Thanks!

**Pull request closed without merging**